### DISCUSSION TITLE
Labels for Project Keys

### DISCUSSION THREAD

**Discussion by axmc1** [deta/cloud-docs#499] [Ideas] [answered]
The title explains it, some kind of labels for project keys, remembering them by some 6 char string kinda messes with brain, etc.

(Sorry for not being formal)

**Comment by AyushSehrawat** [comment]
Yeah, @abdelhai please look into this 🚀

**Comment by abdelhai** [comment]
hi, i don't really understand the concept. could you explain it a bit and/or show how it would work in code/ui.

**Reply by AyushSehrawat** [reply]
@abdelhai Currently when we create the keys, some random keys are shown after we copy and close that pop up which contains the project key.

So, if you get into a situation to delete the keys, you either have to see the project key ( first part before _ ) to know it and delete it or remember it.

So it would be more convenient if project keys could have some label

`Development Key : uhsfos`

**Reply by AyushSehrawat** [reply]
there's another way to remember by the order of how you created them, but labels would be better ig

**Reply by jnsougata** [reply]
In simple terms they are asking for custom key `descriptions` so that they can manage those keys easily in future. As in recent behavior the key has random id which makes it very hard to remember the purpose of the key is made for… if there’s multiple keys. Somewhere I felt it too. There are some unused keys I want to remove but I can’t …as that might break some project unknowingly as I can’t remember why I created the key just by looking at the id.

@abdelhai

**Reply by Sokhavuth** [reply]
I think the word "key" in Deta Base means "UUID" in other database such as MongoDB for example that use _id to store auto-generated uuid. Deta Base is a large scale database that could store millions of documents or records. So it's hard to remember which record with which key was created for which reason. It is not like Redis database that is mainly used to store few session data in memory; and those entries are referred to by "key" or "label". However, it very easy to create a label in a document or record in Deta Base, just add a new key-value pair into the document or record you want to name it.
```js
document = {
    "title": "post_title",
    "content": "post_content",
    "label": "anything_you_want",
}
```

**Reply by AyushSehrawat** [reply]
> I think the word "key" in Deta Base means "UUID" in other database such as MongoDB for example that use _id to store auto-generated uuid. Deta Base is a large scale database that could store millions of documents or records. So it's hard to remember which record with which key was created for which reason. It is not like Redis database that is mainly used to store few session data in memory; and those entries are referred to by "key" or "label". However, it very easy to create a label in a document or record in Deta Base, just add a new key-value pair into the document or record you want to name it.
> 
> ```js
> document = {
>     "title": "post_title",
>     "content": "post_content",
>     "label": "anything_you_want",
> }
> ```

He's talking about project keys, not base

**Comment by abdelhai** [comment, marked as answer]
ah i get it now. i thought we had that. cc @betahuhn

**Reply by BetaHuhn** [reply]
Nope, currently you can't customise the project key name or description. It gets randomly generated which as the others mentioned makes it hard to know which key is used for what.

@Zykkl @AyushSehrawat @jnsougata thanks for the feedback – noted!

**Comment by axmc1** [comment]
Was away for long, thx for clearing it @AyushSehrawat and @jnsougata
